amads/core/vectors_sets.py

```diff
@@ -0,0 +1,294 @@
+# coding: utf-8
+"""
+Basic, shared functionality for sets and vectors.
+We aim for clear and consistent use of "set" and "vector" in  naming and documentation,
+though we tend to use basic classes such as lists in preference over the Python multiset class, for instance.
+
+Argument names are "set" or "vector" where that is the clear expectation and reason for the function.
+Where the function could be more widely applied, arguments are named `input_list`.
+"""
+
+from __future__ import annotations
+
+__author__ = "Mark Gotham"
+
+
+def set_to_vector(
+    input_set: list[int] | tuple[int],
+    min_index: int | None = 0,
+    max_index: int | None = 6,
+) -> tuple:
+    """
+    Converts any "set" (list of integers) into a "vector" (count of integers organised by index).
+    See the paper for full definitions.
+    This is similar to the collections.Counter function, simply returning an ordered list instead of a dict.
+
+    Parameters
+    ----------
+    input_set: The input integers.
+    min_index: The minimum index to use. Defaults to 0. Use 1 for interval vectors, to exlcude 0.
+    max_index: The maximum index to use. Defaults to 6 for interval vectors.
+
+    Returns
+    -------
+    tuple: The corresponding vector.
+
+    Examples
+    --------
+    >>> test_set = [1, 2, 3]
+    >>> set_to_vector(test_set, min_index=0, max_index=6)
+    (0, 1, 1, 1, 0, 0, 0)
+
+    >>> set_to_vector(test_set, min_index=1, max_index=6)
+    (1, 1, 1, 0, 0, 0)
+
+    >>> set_to_vector(test_set, max_index=None)
+    (0, 1, 1, 1)
+
+    """
+    if not max_index:
+        max_index = max(input_set)
+    counts = [0] * (max_index + 1)
+    for item in input_set:
+        counts[item] += 1
+
+    if min_index > 0:
+        counts = counts[min_index:]
+
+    return tuple(counts)
+
+
+def vector_to_set(vector: list[int] | tuple[int]) -> tuple:
```

Comment:
It feels a bit strange that the set is not a set. Consider using a Python set as the output type?